daemon/CryptoImpl/crypto.cpp

```diff
@@ -323,6 +365,53 @@ void Daemon::ApiImpl::CryptoRequestQueue::handlePendingRequest(
             }
             break;
         }
+        case GenerateRandomDataRequest: {
+            qCDebug(lcSailfishCryptoDaemon) << "Handling GenerateRandomDataRequest from client:" << request->remotePid << ", request number:" << request->requestId;
+            QByteArray randomData;
+            quint64 numberBytes = request->inParams.size() ? request->inParams.takeFirst().value<quint64>() : 0;
+            QString csprngEngineName = request->inParams.size() ? request->inParams.takeFirst().value<QString>() : QString();
+            QString cryptosystemProviderName = request->inParams.size() ? request->inParams.takeFirst().value<QString>() : QString();
+            Result result = m_requestProcessor->generateRandomData(
```

Comment:
We need to fix the way we marshal parameters internally a bit better, definitely (I think we need per-request-type structs, to avoid this list-style procedure).  But its important to note that the fallback cases here should never occur in practice (so we should add Q_ASSERTs to check them), the default constructed values are there solely to avoid a crash in case we missed some codepath in our unit tests which gets hit in the wild.  Yes, we could and should return an error immediately; but let's do that at the same time we fix the marshalling, IMO.

Reply:
Ok.